src/client/main.go

```diff
@@ -72,52 +75,153 @@ var CliClientOpenCommand = cli.Command{
 			BrowserScriptPath: fmt.Sprintf("/tmp/nvrh-browser-%s", sessionId),
 		}
 
-		// TODO Could really use a context to pass around instead of a bunch of
-		// args.
-		// socketPath := fmt.Sprintf("/tmp/nvrh-socket-%d", sessionId)
 		if nvrhContext.ShouldUsePorts {
 			min := 1025
 			max := 65535
-			nvrhContext.PortNumber = rand.IntN((max - min) + min)
+			nvrhContext.PortNumber = rand.IntN(max-min) + min
 		}
 
 		if nvrhContext.Server == "" {
 			return fmt.Errorf("<server> is required")
 		}
 
-		go ssh_helpers.StartRemoteNvim(nvrhContext)
+		var nv *nvim.Nvim
 
+		doneChan := make(chan error)
+
+		signalChan := make(chan os.Signal, 1)
+		signal.Notify(signalChan, os.Interrupt)
+
+		// Prepare remote instance.
+		go func() {
+			remoteCmd := ssh_helpers.BuildRemoteNvimCmd(&nvrhContext)
+			nvrhContext.CommandsToKill = append(nvrhContext.CommandsToKill, remoteCmd)
+
+			// We don't want the ssh process ending too early, if it does we can't
+			// clean up the remote nvim instance.
+			exec_helpers.PrepareForForking(remoteCmd)
```

Comment:
Obvious in hindsight, but this breaks logging in via password since the "enter password" bit isn't shown in a terminal.

:(